EIPS/eip-7756.md

```diff
@@ -0,0 +1,177 @@
+---
+eip: 7756
+title: EOF/EVM Trace Specification
+description: Updates EIP-3155 tracing to add EOF support
+author: Martin Holst Swende (@holiman), Marius van der Wijden (@MariusVanDerWijden), Danno Ferrin (@shemnon)
+discussions-to: https://ethereum-magicians.org/t/eip-7756-eof-evm-trace-specification/20806
+status: Draft
+type: Standards Track
+category: Interface
+created: 2024-08-13
+requires: 3155
+---
+
+## Abstract
+
+Updates the [EIP-3155](./eip-3155.md) JSON tracing specification to support EOF features and light tracing.
+
+## Motivation
+
+EIP-3155 defined a tracing standard for Legacy EVM operations. However, the EVM Object Format ([EIP-7692](./eip-7692.md)) adds a number of features that need to be reflected in debugging traces.
+
+The use of these traces has also moved out from state testing, including live block tracing and differential fuzzing, increasing the need to keep tracing up to date.
+
+This EIP has multiple goals:
+
+- Add members to the trace object to support new EOF features.
+- Support tracing contracts contained in an EOF container as well as uncontained "legacy" contracts in the same trace.
+- Clarify any previous ambiguities in the EIP-3155 specification.
+
+## Specification
+
+To promote clarity and provide a cohesive specification, the entire tracing specification will be presented with alterations in-line rather than as a set of diffs on top of EIP-3155. Differences will be highlighted in the Backwards Compatibility section.
+
+### Datatypes
+
+| Type       | Explanation                                                    | Example                              |
+|------------|----------------------------------------------------------------|--------------------------------------|
+| Number     | Plain json number                                              | "pc":0                               |
+| Hex-Number | Hex-encoded number                                             | "gas":"0x2540be400"                  |
+| String     | Plain string                                                   | "opName":"PUSH1"                     |
+| Hex-String | Hex-encoded string                                             | "returnData":"0x60616203"            |
+| Array      | Array of Hex-Strings                                           | "stack":["0x0", "0x0"]               |
+| Key-Value  | Key-Value structure with key and values encoded as hex strings | "storage":{"0x0":"0x1", "0x1":"0x1"} |
+| Boolean    | Json bool can either be true or false                          | "pass": true                         |
+
+### Output
+
+- The client outputs one JSON object per EVM operation executed.
+- The client MUST NOT output multiple lines for the same operation execution.
+- The client MUST NOT output a line for the `STOP` operation if an error occurred, or if the contract runs out of instructions.
+
+Each object contains the following members.
```

Comment:
Since these are JSON objects, would you consider using [JSON Schema](https://json-schema.org/) to describe the format? The schema could also be included as an asset instead.

Reply:
You can define only the properties introduced in this proposal and leave the rest undefined. For example:

```json
{
    "$schema": "https://json-schema.org/draft/2019-09/schema",
    "type": "object",
    "properties": {
      "section": {
        "type": "number"
      },
      "immediate": {
        "$ref": "#/$defs/hex-string"
      },
      "functionDepth": {
        "type": "number"
      }
    },
    "$defs": {
      "hex-string": {
        "type": "string",
        "pattern": "^0x[0-9A-Fa-f]*$",
      }
    }
}
```